### PR TITLE
Pattern format and indentation work

### PR DESCRIPTION
**Pull request by jaredpar**
This updates the workspace layer to understand the basic formatting and indentation logic for patterns.  In particular the work surrounds property patterns:

```
object o = null;
if (o is Point { X is 42, Y is var y })
```

I patterned the expected behavior off of collection initializers / anonymous object creation.  They are the closest features syntactically to this.  

After this change I'm able to type patterns quite comfortably in the editor.  Probably some more advanced scenarios i'm missing so please do point those out.

### CODE REVIEW

**Comment by jaredpar** [comment]
CC @gafter, @CyrusNajmabadi, @AlekseyTs, @balajikris, @basoundr

**Comment by jaredpar** [comment]
Resolves #9071, #9078

**Comment by jaredpar** [comment]
Resolves #9075 as well now.

File: src/Workspaces/CoreTest/FormattingTests.cs [hunk 1]
```diff
@@ -7,6 +8,7 @@
 using Xunit;
 using CS = Microsoft.CodeAnalysis.CSharp;
 using VB = Microsoft.CodeAnalysis.VisualBasic;
+using System;
```
Comment (CyrusNajmabadi) [inline src/Workspaces/CoreTest/FormattingTests.cs]: undo these changes.

**Comment by CyrusNajmabadi** [comment]
:+1:
